homeassistant/components/homekit_controller/climate.py

```diff
@@ -170,8 +178,49 @@ def get_characteristic_types(self) -> list[str]:
             CharacteristicsTypes.TEMPERATURE_COOLING_THRESHOLD,
             CharacteristicsTypes.TEMPERATURE_HEATING_THRESHOLD,
             CharacteristicsTypes.SWING_MODE,
+            CharacteristicsTypes.ROTATION_SPEED,
         ]
 
+    @property
+    def fan_modes(self) -> list[str] | None:
+        """Return the available fan modes."""
+        if self.service.has(CharacteristicsTypes.ROTATION_SPEED):
+            return [FAN_OFF,FAN_LOW, FAN_MEDIUM, FAN_HIGH]
+        return None
+
+    @property
+    def fan_mode(self) -> str | None:
+        """Return the current fan mode."""
+        speed = self.service.value(CharacteristicsTypes.ROTATION_SPEED)
```

Comment:
Rotation speed needs to be scaled to min and max value

The HomeKit model allows the device to set almost any min and max

The min could be 100 and the max could be 10000 and iOS would show 1000 as 10%

Reply:
done